archive/proxy_compat.go

```diff
@@ -0,0 +1,56 @@
+package archive
+
+import (
+	"fmt"
+
+	"github.com/lirm/aeron-go/aeron/atomic"
+	"github.com/lirm/aeron-go/archive/codecs"
+)
+
+// ArchiveId Get the id of the archive
+func (proxy *Proxy) ArchiveId(correlationId, controlSessionId int64) (bool, error) {
+	bytes, err := codecs.ArchiveIdPacket(proxy.marshaller, proxy.rangeChecking, controlSessionId, correlationId)
+	if err != nil {
+		return false, err
+	}
+
+	ret := proxy.Offer(atomic.MakeBuffer(bytes, len(bytes)), 0, int32(len(bytes)), nil)
+	if ret < 0 {
+		return false, fmt.Errorf("Offer failed: %d", ret)
+	}
+
+	return ret > 0, nil
```

Comment:
this one i am not 100% sure. 

Generally in the codebase proxy.Offer() is usually check >= 0 where 0 is a valid position. Java version though is checked against > 0